hnn_core/cells_default.py

```diff
@@ -225,77 +371,13 @@ def pyramidal(cell_name, pos=(0, 0, 0), override_params=None, gid=None):
         The GID is an integer from 0 to n_cells, or None if the cell is not
         yet attached to a network. Once the GID is set, it cannot be changed.
     """
-
-    if cell_name == 'L5Pyr':
-        p_all_default = get_L5Pyr_params_default()
-        # units = ['pS/um^2', 'S/cm^2', 'pS/um^2', '??', 'tau', '??']
-        mechanisms = {
-            'hh2': ['gkbar_hh2', 'gnabar_hh2',
-                    'gl_hh2', 'el_hh2'],
-            'ca': ['gbar_ca'],
-            'cad': ['taur_cad'],
-            'kca': ['gbar_kca'],
-            'km': ['gbar_km'],
-            'cat': ['gbar_cat'],
-            'ar': ['gbar_ar']
-        }
-        section_names = ['apical_trunk', 'apical_1',
-                         'apical_2', 'apical_tuft',
-                         'apical_oblique', 'basal_1', 'basal_2', 'basal_3']
-        sec_pts, topology = _secs_L5Pyr()
-    elif cell_name == 'L2Pyr':
-        p_all_default = get_L2Pyr_params_default()
-        mechanisms = {
-            'km': ['gbar_km'],
-            'hh2': ['gkbar_hh2', 'gnabar_hh2',
-                    'gl_hh2', 'el_hh2']}
-        section_names = ['apical_trunk', 'apical_1', 'apical_tuft',
-                         'apical_oblique', 'basal_1', 'basal_2', 'basal_3']
-        sec_pts, topology = _secs_L2Pyr()
+    if cell_name == 'L2Pyr':
+        return _cell_L2Pyr(override_params, pos=pos, gid=gid)
+    elif cell_name == 'L5Pyr':
+        return _cell_L5Pyr(override_params, pos=pos, gid=gid)
```

Comment:
What is the purpose of the `pyramidal()` function now? We may not want to get rid of it here to prevent the diff from getting too big, but I don't see the point in having the added level of abstraction without any benefit.

Reply:
agreed!